plugins/modules/clustering/consul/consul_role.py

```diff
@@ -0,0 +1,476 @@
+        self.name = input["name"]
+        self.datacenter = input["datacenter"] if "datacenter" in input else None
+
+    def to_dict(self):
+        return {
+            "NodeName": self.name,
+            "Datacenter": self.datacenter
+        }
+
+
+class RoleLink:
+    def __init__(self, dict):
+        self.id = dict.get("id", None)
+        self.name = dict.get("name", None)
+
+    def to_dict(self):
+        return {
+            "ID": self.id,
+            "Name": self.name
+        }
+
+
+class PolicyLink:
+    def __init__(self, dict):
+        self.id = dict.get("id", None)
+        self.name = dict.get("name", None)
+
+    def to_dict(self):
+        return {
+            "ID": self.id,
+            "Name": self.name
+        }
+
+
+class Configuration:
+    """
+    Configuration for this module.
+    """
+
+    def __init__(self, token=None, host=None, scheme=None, validate_certs=None, name=None, description=None, port=None,
+                 policies=None, service_identities=None, node_identities=None, state=None):
+        self.token = token                                    # type: str
+        self.host = host                                                            # type: str
+        self.port = port                                                            # type: int
+        self.scheme = scheme                                                        # type: str
+        self.validate_certs = validate_certs                                        # type: bool
+        self.name = name                                                            # type: str
+        self.description = description                                              # type: str
+        self.policies = [PolicyLink(p) for p in policies]                           # type: list(PolicyLink)
+        self.service_identities = [ServiceIdentity(s) for s in service_identities]  # type: list(ServiceIdentity)
+        self.node_identities = [NodeIdentity(n) for n in node_identities]           # type: list(NodeIdentity)
+        self.state = state                                                          # type: str
+
+
+class Output:
+    """
+    Output of an action of this module.
+    """
+
+    def __init__(self, changed=None, operation=None, role=None):
+        self.changed = changed      # type: bool
+        self.operation = operation  # type: str
+        self.role = role          # type: dict
+
+
+def check_dependencies():
+    """
+    Checks that the required dependencies have been imported.
+    :exception ImportError: if it is detected that any of the required dependencies have not been imported
+    """
+
+    if not has_requests:
+        raise ImportError(
+            "requests required for this module. See https://pypi.org/project/requests/")
+
+
+def main():
+    """
+    Main method.
+    """
+    module = AnsibleModule(_ARGUMENT_SPEC, supports_check_mode=False)
```

Comment:
Generally modules should support check mode if possible. Usually implementing check mode is pretty easy, as you need to do idempotency checks anyway. (The main reason to not support check mode is when idempotency checks are not done by the module, but by the service/software called by the module.)